src/ln/channelmanager.rs

```diff
@@ -5362,6 +5365,414 @@ mod tests {
 		claim_payment(&nodes[0], &[&nodes[1]], payment_preimage_2);
 	}
 
+	#[test]
```

Comment:
Maybe add a test where between Alice and Caroll, Bob knows a PermanentFailure and so fail backward htlcs ? Do we fail backward if we have TemporaryFailure on outgoing channel and it's long enough to timeout an inbound HTLC on ingoing channel ?

Reply:
Hmm, not sure what you're asking for? If you're asking about a failed attempt to relay an HTLC where the new commitment tx generation monitor update fails to commit thats still unimplemented!() so its not testable :p (and the back fail isnt written for that yet, either).

We should back-fail on TemporaryFailure via the would_broadcast_at_height mechanism/on-chain confirmation just like normal, but the last step isnt implemented yet.

Reply:
Okay noted not implemented or we go onchain in this case.